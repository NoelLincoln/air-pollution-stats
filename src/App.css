@font-face {
  font-family: 'Gillsans';
  src: url('./fonts/gill-sans/gillsans.ttf') format('truetype');
  font-weight: normal;
  font-style: normal;
}

@font-face {
  font-family: 'Lato';
  src: url('./fonts/lato/Lato-Regular.ttf') format('truetype');
  font-weight: normal;
  font-style: normal;
}

* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
  font-family: 'Lato', sans-serif;
  color: white;
}

body {
  background-color: #fe5294;
  color: #fff;
}

p {
  font-family: 'Gillsans', sans-serif;
}

.nav {
  display: flex;
  justify-content: space-between;
  position: sticky;
  top: 0;
  padding: 1rem 2rem 1rem 2rem;
  background-color: #bb4b76;
}

.App-header {
  text-align: center;
  color: #fff;
}

.country-img {
  display: flex;
  align-items: center;
  border: 5px solid transparent;
}

.world-img > img {
  width: 200px;
  height: 200px;
}

.state-img {
  width: 150px;
  height: 100px;
}

.list-countries {
  display: flex;
  flex-direction: column;
  gap: 0.5rem;
  height: 16rem;
  overflow-y: scroll;
}

button {
  background-color: #d28fc2;
  border: none;
  border-radius: 5px;
  color: white;
  text-align: center;
  text-decoration: none;
  display: inline-block;
  font-size: 16px;
}

.list-countries button {
  width: 80%;
  margin: auto;
}

.country-details {
  display: flex;
  flex-direction: column;
  gap: 2rem;
  align-items: center;
  justify-content: center;
  width: 50%;
  height: 200px;
  margin: auto;
  text-align: center;
  border-left: 1px solid;
  background-color: #fe5294;
}

.country-details span {
  border-bottom: 1px solid;
}

.state-list-container {
  background-color: #bb4b76;
}

.state-list,
.air-components,
.search-box {
  display: flex;
  width: 100%;
  flex-wrap: wrap;
}

.air-components {
  justify-content: center;
  gap: 1rem;
  margin-top: 2rem;
}

.state-list > li {
  flex: 0 0 calc(50% - 16px); /* Adjust the width as needed */
  background-color: #aa4a6f;
  margin-bottom: 16px; /* Adjust the margin as needed */
}

.state-list li:nth-child(4n - 3),
.state-list li:nth-child(4n) {
  background-color: #bb4b76;
}

.state-stat {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  background-color: #aa4a6f;
  width: 200px;
}

.arrow-right-cont {
  width: 100%;
  display: flex;
  justify-content: flex-end;
  padding: 5px;
}

.arrow-right-cont > img {
  width: 15px;
  height: 15px;
}

.state-pollution-container {
  display: flex;
  flex-direction: column;
  gap: 1.5rem;
  margin-top: 5rem;
  cursor: pointer;
}

.state-pollution-data {
  display: flex;
  width: 100%;
  background-color: darkgray;
  height: 5rem;
  align-items: center;
  justify-content: center;
}

.air-components-stat {
  display: flex;
  width: 40%;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  background-color: #aa4a6f;
  height: 8rem;
}

.change-country-btn {
  background-color: #d28fc2;
  border: none;
  border-radius: 5px;
  color: white;
  padding: 10px 5px;
  text-align: center;
  text-decoration: none;
  display: inline-block;
  font-size: 12px;
}

.search-box {
  padding: 1rem 1.5rem;
  align-items: center;
}

input[type='text'] {
  width: 100%;
  padding: 12px 20px;
  margin: 8px 0;
  display: inline-block;
  border: 1px solid #ccc;
  border-radius: 4px;
  box-sizing: border-box;
  color: black;
}

.state-air-details {
  display: flex;
  flex-direction: column;
  margin-top: 1rem;
  gap: 0.5rem;
}

.state-air-info {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.air-quality-index {
  display: flex;
  align-items: baseline;
  justify-content: center;
  gap: 2rem;
  margin: 2rem 0;
}

h3 {
  padding-left: 2rem;
}

.arrow-back img {
  width: 14px;
}

.menu-icon > img {
  width: 14px;
}

.air-quality-guide-container {
  display: flex;
  flex-direction: column;
  margin-top: 2rem;
  align-items: center;
  border-top: 1px solid;
  width: 100%;
  padding: 1rem;
}

.air-quallity-index-guide {
  list-style-type: none;
}

.statenamelist {
  display: flex;
  flex-direction: column;
  align-items: flex-end;
}
