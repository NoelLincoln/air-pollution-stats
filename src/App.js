import './App.css';

function App() {
  return (
    <div className="App">
      <header className="App-header">
        <h1>2023 stats</h1>
      </header>
    </div>
  );
}

export default App;
