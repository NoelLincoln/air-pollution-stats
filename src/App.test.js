import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store'; // Import redux-mock-store
import App from './App';

const mockStore = configureStore([]);
const initialState = {
  latitude: -1.3025617,
  longitude: 36.9916181,
  countries: [],
  country: 'Kenya',
  countryShortCode: 'KE',
  image: 'https://twemoji.maxcdn.com/2/svg/1f1f0-1f1ea.svg',
  states: [
    {
      id: 64772,
      name: 'Ahero',
    },
    {
      id: 64773,
      name: 'Athi River',
    },
  ],
  totalNumberOfStates: 110,
  selectedState: [],
  airQualityData: null,
  isDataFetched: true,
  fetchStatus: 'fulfilled',
  error: null,
};

const store = mockStore(initialState);

test('App component renders correctly', () => {
  const { asFragment } = render(
    <Provider store={store}>
      <BrowserRouter>
        <App />
      </BrowserRouter>
    </Provider>,
  );

  expect(asFragment()).toMatchSnapshot();
});
