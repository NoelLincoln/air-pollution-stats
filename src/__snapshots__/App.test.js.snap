// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders correctly 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      class="nav"
    >
      <div
        class="empty-div"
      />
      <div>
        <h2>
          Air Pollution Stats
        </h2>
      </div>
      <div
        class="menu-icon"
      >
        <img
          alt="Menu"
          src="menu-icon.png"
        />
      </div>
    </div>
    <section
      class="home-container"
    >
      <div
        class="country"
      >
        <div
          class="country-img"
        >
          <div
            class="world-img"
          >
            <img
              alt=""
              src="world-globe.gif"
            />
          </div>
          <div
            class="country-details"
          >
            <h1 />
            <span>
              0
              <br />
              States
            </span>
            <button
              class="change-country-btn"
              type="button"
            >
              Change country
            </button>
          </div>
        </div>
      </div>
    </section>
    <section
      class="state-list-container"
    >
      <div
        class="search-box"
      >
        <h4>
          Stats By State
        </h4>
        <input
          placeholder="Search for a state"
          type="text"
          value=""
        />
        <ul />
      </div>
      <ul
        class="state-list"
      />
    </section>
  </div>
</DocumentFragment>
`;
